safehttp/plugins/coop/coop.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package coop provides Cross-Origin-Opener-Policy protection. Specification: https://html.spec.whatwg.org/#cross-origin-opener-policies
+package coop
+
+import (
+	"github.com/google/go-safeweb/safehttp"
+)
+
+// Mode represents a COOP mode.
+type Mode string
+
+const (
+	// SameOrigin is the strictest and safest COOP available: windows can keep a reference of windows they open only if they are same-origin.
+	SameOrigin Mode = "same-origin"
+	// SameOriginAllowPopups relaxes the same-origin COOP: windows on this origin that open other windows are allowed to keep a reference, but the opposite is not valid.
+	SameOriginAllowPopups Mode = "same-origin-allow-popups"
+	// UnsafeNone disables COOP: this is the default value in browsers.
+	UnsafeNone Mode = "unsafe-none"
+)
+
+// Policy represents a Cross-Origin-Opener-Policy value.
+type Policy struct {
+	// Mode is the mode for the policy.
+	Mode Mode
+	// ReportingGroup is an optional reporting group that needs to be defined with the Reporting API.
+	ReportingGroup string
+	// ReportOnly makes the policy report-only if set.
+	ReportOnly bool
+}
+
+func (p Policy) String() string {
```

Comment:
Can you add a brief comment to this method since it's exported? Same for `Before`, `Commit` and `Match`